public/pages/DetectorDetail/containers/DetectorDetail.tsx

```diff
@@ -0,0 +1,340 @@
+                    {detector.enabledTime
+                      ? moment(detector.enabledTime).format('MM/DD/YY h:mm a')
+                      : '?'}
+                  </EuiHealth>
+                ) : (
+                  <EuiHealth color="subdued">
+                    {detector.featureAttributes &&
+                    detector.featureAttributes.length
+                      ? detector.disabledTime
+                        ? `Stopped at ${moment(detector.disabledTime).format(
+                            'MM/DD/YY h:mm a'
+                          )}`
+                        : 'Detector is not started'
+                      : 'Feature required to start the detector'}
+                  </EuiHealth>
+                )}
+              </h1>
+            </EuiTitle>
+          </EuiFlexItem>
+
+          <EuiFlexItem grow={false}>
+            <DetectorControls
+              onEditDetector={() => {
+                detector.enabled
+                  ? setShowStopDetectorModalFor('detector')
+                  : props.history.push(`/detectors/${detectorId}/edit`);
+              }}
+              onDelete={() => setShowDeleteDetectorModal(true)}
+              onStartDetector={() => handleStartAdJob(detectorId)}
+              onStopDetector={() =>
+                monitor
+                  ? setShowMonitorCalloutModal(true)
+                  : handleStopAdJob(detectorId)
+              }
+              onEditFeatures={() => {
+                detector.enabled
+                  ? setShowStopDetectorModalFor('features')
+                  : props.history.push(`/detectors/${detectorId}/features`);
+              }}
+              detector={detector}
+            />
+          </EuiFlexItem>
+        </EuiFlexGroup>
+
+        <EuiFlexGroup>
+          <EuiFlexItem>
+            <EuiTabs>
+              {tabs.map(tab => (
+                <EuiTab
+                  onClick={() => {
+                    handleTabChange(tab.route);
+                  }}
+                  isSelected={tab.id === selectedTab}
+                  key={tab.id}
+                >
+                  {tab.name}
+                </EuiTab>
+              ))}
+            </EuiTabs>
+          </EuiFlexItem>
+        </EuiFlexGroup>
+      </EuiFlexGroup>
+      {showDeleteDetectorModal ? (
+        <EuiOverlayMask>
+          <ConfirmModal
+            title="Delete detector?"
+            description="Detector and feature configuration will be removed and the deletion is irreversible and you can’t review anomaly result of this detector on Kibana. To confirm deletion, click the delete button."
+            callout={monitorCallout}
+            confirmButtonText="Delete"
+            confirmButtonColor="danger"
+            onClose={() => setShowDeleteDetectorModal(false)}
+            onCancel={() => setShowDeleteDetectorModal(false)}
+            onConfirm={handleDelete}
+          />
+        </EuiOverlayMask>
+      ) : null}
```

Comment:
You could have single state which would also sets the other props which can reduce the repeatable code.

```
const [detailsModel, setDetailsModel] = useState({
modelType:
title: '',
description: '',
confirmButton: '',
....other required props.
})
```

Reply:
Good suggestion, will change